src/agentTest.ts

```diff
@@ -0,0 +1,360 @@
+import { type AvailableDefinition, type AgentTestConfig, type AiEvaluationDefinition, type TestSpec } from './types.js';
+
+Messages.importMessagesDirectory(__dirname);
+const messages = Messages.loadMessages('@salesforce/agents', 'agentTest');
+
+/**
+ * Events emitted during agent test creation for consumers to listen to and keep track of progress.
+ */
+export const AgentTestCreateLifecycleStages = {
+  CreatingLocalMetadata: 'Creating Local Metadata',
+  Waiting: 'Waiting for the org to respond',
+  DeployingMetadata: 'Deploying Metadata',
+  Done: 'Done',
+};
+
+/**
+ * A client side representation of an agent test (AiEvaluationDefinition) within an org.
+ * Also provides utilities such as creating and listing agent tests, and converting between
+ * agent test spec and AiEvaluationDefinition.
+ *
+ * **Examples**
+ *
+ * Create a new instance from an agent test spec:
+ *
+ * `const agentTest = new AgentTest({ specPath: path/to/specfile });`
+ *
+ * Get the metadata content of an agent test:
+ *
+ * `const metadataContent = await agentTest.getMetadata();`
+ *
+ * Write the metadata content to a file:
+ *
+ * `await agentTest.writeMetadata('path/to/metadataFile');`
+ */
+export class AgentTest {
+  private specData?: TestSpec;
+  private data?: AiEvaluationDefinition;
+
+  /**
+   * Create an AgentTest based on one of:
+   *
+   * 1. AiEvaluationDefinition API name.
+   * 2. Path to a local AiEvaluationDefinition metadata file.
+   * 3. Path to a local agent test spec file.
+   *
+   * @param config AgentTestConfig
+   */
+  public constructor(private config: AgentTestConfig) {
+    const { name, mdPath, specPath } = config;
+
+    if (!name && !mdPath && !specPath) {
+      throw messages.createError('invalidAgentTestConfig');
+    }
+  }
+
+  /**
+   * List the AiEvaluationDefinitions available in the org.
+   */
+  public static async list(connection: Connection): Promise<AvailableDefinition[]> {
+    return connection.metadata.list({ type: 'AiEvaluationDefinition' });
+  }
+
+  /**
+   * Creates and deploys an AiEvaluationDefinition from a specification file.
+   *
+   * @param apiName - The API name of the AiEvaluationDefinition to create
+   * @param specFilePath - The path to the specification file to create the definition from
+   * @param options - Configuration options for creating the definition
+   * @param options.outputDir - The directory where the AiEvaluationDefinition file will be written
+   * @param options.preview - If true, writes the AiEvaluationDefinition file to <api-name>-preview-<timestamp>.xml in the current working directory and does not deploy it
+   *
+   * @returns Promise containing:
+   * - path: The filesystem path to the created AiEvaluationDefinition file
+   * - contents: The AiEvaluationDefinition contents as a string
+   * - deployResult: The deployment result (if not in preview mode)
+   *
+   * @throws {SfError} When deployment fails
+   */
+  public static async create(
```

Comment:
NIT - only since the jsdoc is so nice : but it's missing the connection param description